applications/TopologyOptimizationApplication/custom_strategies/custom_schemes/residualbased_incrementalupdate_static_simp_scheme.h

```diff
@@ -0,0 +1,268 @@
+
+
+namespace Kratos
+{
+
+/**@name Kratos Globals */
+/*@{ */
+
+
+/*@} */
+/**@name Type Definitions */
+/*@{ */
+
+/*@} */
+
+
+/**@name  Enum's */
+/*@{ */
+
+
+/*@} */
+/**@name  Functions */
+/*@{ */
+
+
+/*@} */
+/**@name Kratos Classes */
+/*@{ */
+
+template<class TSparseSpace,class TDenseSpace >
+class ResidualBasedIncrementalUpdateStaticSIMPScheme : public ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace,TDenseSpace>
+{
+
+public:
+    /**@name Type Definitions */
+    /*@{ */
+
+    //typedef boost::shared_ptr< ResidualBasedIncrementalUpdateStaticSIMPScheme<TSparseSpace,TDenseSpace> > Pointer;
+    KRATOS_CLASS_POINTER_DEFINITION( ResidualBasedIncrementalUpdateStaticSIMPScheme );
+
+    typedef ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace,TDenseSpace> BaseType;
+
+    typedef typename BaseType::TDataType TDataType;
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+
+
+    /// The definition of the vector containing the equation ids
+    typedef Element::EquationIdVectorType                              EquationIdVectorType;
+
+    /*@} */
+    /**@name Life Cycle
+    */
+    /*@{ */
+
+    /** Constructor.
+    */
+    ResidualBasedIncrementalUpdateStaticSIMPScheme()
+        : ResidualBasedIncrementalUpdateStaticScheme<TSparseSpace,TDenseSpace>()
+    {}
+
+    /** Destructor.
+    */
+    ~ResidualBasedIncrementalUpdateStaticSIMPScheme() override {} 
+    ///virtual ~ResidualBasedIncrementalUpdateStaticSIMPScheme() {} alte Version 14.12
+
+
+    /*@} */
+    /**@name Operators
+    */
+    /*@{ */
+
+    // ---------------------------------------------------------------------------------------------------------------------------------------------
+    // --------------------------------- UPDATE LHS AND RHS ----------------------------------------------------------------------------------------
+    // ---------------------------------------------------------------------------------------------------------------------------------------------
+
+    /// This function calculates a new Youngs Modulus based on the densities and multiplies it into the
+    /// LHS and RHS contributions of the complete system
+
+
+
+    void CalculateSystemContributions(
+        Element& rCurrentElement,
+        LocalSystemMatrixType& rLHSContribution,
+        LocalSystemVectorType& rRHSContribution,
+        EquationIdVectorType& rEquationId,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override
+    {
+        KRATOS_TRY
+        //Initializing the non linear iteration for the current element
+        rCurrentElement.InitializeNonLinearIteration(rCurrentProcessInfo);
```

Comment:
Thanks for the detailed review @armingeiser! I am not quite sure if I can follow you!

Reply:
In this function, you are multiplying LHS and RHS with a scaling factor based on the penalty term. This could also be done directly in your SIMP element, in the related CalculateLHS/RHS functions. 

Since this modification of the LHS/RHAS is the only thing that happens in this scheme, it would become obsolete and you could use teh scheme this one is inheriting from.

Reply:
I will implement this in a second moment if possible.

Reply:
OK for me.